src/midje/parsing/0_to_fact_form/generative.clj

```diff
@@ -71,19 +73,86 @@
           num-tests               (or (:num-tests opts) 10)
           quick-check-opts        (->> (select-keys opts [:seed :max-size])
                                        (into [])
-                                       flatten)]
-      `(let [fact-fn#       (fn ~prop-names
+                                       flatten)
+          fact-fn-name            (gensym "fact-fn")]
```

Comment:
Why did you lift this out of the quoted syntax?

Reply:
I was trying to make the for-all parser similar to the gen-let parser, in hope of finding common abstractions to share. Not a raging success so far.